### DISCUSSION TITLE
Invalid Signature Issue

### DISCUSSION THREAD

**Discussion by dannyhutch2511** [linux-surface/linux-surface#605] [General]
Tried running the 5.14.11-surface but an error come up with an invalid signature

error: /boot/vmlinuz-5.14.11-surface has invalid signature
error: you need to load the kernel first

New to linux and working on a Surface 3
Currently already running Ubuntu with Linux 5.11.0-37-generic

Any help would be appreciated thanks

**Comment by qzed** [comment]
I assume that the stock Ubuntu kernel boots just fine? In that case you've already enabled "Secure Boot with 3rd party keys" in the UEFI menu (which otherwise is the first step). What you're missing is the key that's used to sign the Surface kernels, which is required for validation and not having that installed is the reason why it prints that message/validation fails. To install that, boot into the stock kernel and install the `linux-surface-secureboot-mok` package. During installation, it will print some instructions which you'll have to follow (reboot, enroll the key, enter `surface` to confirm). If you have already installed that package but didn't follow those steps, you can uninstall and re-install it (not just re-install) to trigger the install script again.

**Comment by dannyhutch2511** [comment]
Hi Maximilian, that worked perfectly for me now thank you

**Comment by dannyhutch2511** [comment]
I have it up and running now but I can't see the battery indicator and was wondering is there a way to get it going? I've installed and acpi but that wasn't able to get it showing.

**Comment by JGuptill** [comment]
related question. this doesn't seem to carry over between surface devices? I'm looking to be able to use a usb-c drive on any surface laptop my company has for troubleshooting or just securely wiping viral systems. same error but on Mint.
